modules/component/src/core/potential-observable.ts

```diff
@@ -1,13 +1,25 @@
-import { from, Observable, ObservableInput, of } from 'rxjs';
+import { from, isObservable, Observable } from 'rxjs';
 
-export type PotentialObservable<T> = ObservableInput<T> | null | undefined;
+export type ObservableOrPromise<T> = Observable<T> | PromiseLike<T>;
+
+export type PotentialObservable<T> = T | ObservableOrPromise<T>;
 
 export function fromPotentialObservable<T>(
   potentialObservable: PotentialObservable<T>
-): Observable<T | null | undefined> {
-  if (!potentialObservable) {
-    return of(potentialObservable as null | undefined);
+): Observable<T> {
+  if (isObservable(potentialObservable)) {
+    return potentialObservable;
+  }
+
+  if (isPromiseLike(potentialObservable)) {
+    return from(potentialObservable);
   }
 
-  return from(potentialObservable);
+  return new Observable((subscriber) => {
+    subscriber.next(potentialObservable);
+  });
```

Comment:
returning observable that never completes instead of `of(potentialObservable)` to avoid emitting of `complete` event for static (non-observable) values that will trigger change detection unnecessarily.

Reply:
This is useful information and can be moved into a code comment.